File: Dijkstra.cs
using System;

class Dijkstra
{
    static int INF = int.MaxValue;

    static int[] DijkstraAlgo(int[,] graph, int start)
    {
        int n = graph.GetLength(0);
        int[] dist = new int[n];
        bool[] visited = new bool[n];

        // Initialisation
        for (int i = 0; i < n; i++)
        {
            dist[i] = INF;
            visited[i] = false;
        }

        dist[start] = 0;

        for (int count = 0; count < n - 1; count++)
        {
            int min = INF;
            int u = -1;

            // Trouver le sommet avec la distance minimale
            for (int i = 0; i < n; i++)
            {
                if (!visited[i] && dist[i] <= min)
                {
                    min = dist[i];
                    u = i;
                }
            }

            if (u == -1)
                break;

            visited[u] = true;

            // Mise à jour des distances
            for (int v = 0; v < n; v++)
            {
                if (!visited[v] &&
                    graph[u, v] != 0 &&
                    dist[u] != INF &&
                    dist[u] + graph[u, v] < dist[v])
                {
                    dist[v] = dist[u] + graph[u, v];
                }
            }
        }

        return dist;
    }

    static void Main()
    {
        int[,] graph = {
            { 0, 4, 0, 0, 8 },
            { 4, 0, 3, 0, 2 },
            { 0, 3, 0, 7, 0 },
            { 0, 0, 7, 0, 1 },
            { 8, 2, 0, 1, 0 }
        };

File: Dijkstra.php
<?php

define("INF", PHP_INT_MAX);

function dijkstra($graph, $start) {
    $n = count($graph);
    $dist = array_fill(0, $n, INF);
    $visited = array_fill(0, $n, false);

    $dist[$start] = 0;

    for ($i = 0; $i < $n - 1; $i++) {
        $min = INF;
        $u = -1;

        // Trouver le sommet non visité avec la distance minimale
        for ($j = 0; $j < $n; $j++) {
            if (!$visited[$j] && $dist[$j] < $min) {
                $min = $dist[$j];
                $u = $j;
            }
        }

        if ($u == -1) break;

        $visited[$u] = true;

        // Mise à jour des distances
        for ($v = 0; $v < $n; $v++) {
            if (
                !$visited[$v] &&
                $graph[$u][$v] != 0 &&
                $dist[$u] != INF &&
                $dist[$u] + $graph[$u][$v] < $dist[$v]
            ) {
                $dist[$v] = $dist[$u] + $graph[$u][$v];
            }
        }
    }

    return $dist;
}

// Exemple d'utilisation
$graph = [
    [0, 5, 0, 0, 2],
    [5, 0, 3, 0, 0],
    [0, 3, 0, 7, 0],
    [0, 0, 7, 0, 4],
    [2, 0, 0, 4, 0]
];

$start = 0;
$distances = dijkstra($graph, $start);

echo "Distances depuis le sommet $start\n";
foreach ($distances as $i => $d) {
    echo "Vers $i : $d\n";
}
